freshmaker/handlers/botas/botas_shipped_advisory.py

```diff
@@ -660,9 +660,13 @@ def _completion_with_previous_rebuilds(self, nvrs_mapping: dict[str, str]) -> di
             while former_build:
                 nvr_mappings_to_add[former_build.rebuilt_nvr] = latest_nvr
                 log.info(f"Found former rebuild '{former_build.rebuilt_nvr}' of '{latest_nvr}'")
+                later_build = former_build
                 former_build = ArtifactBuild.query.filter_by(
-                    rebuilt_nvr=former_build.original_nvr
+                    rebuilt_nvr=later_build.original_nvr
                 ).first()
+                # Add most original NVR to the mapping as well
+                if not former_build:
+                    nvr_mappings_to_add[later_build.original_nvr] = latest_nvr
```

Comment:
This can be simplified by:
```
for original_nvr, latest_nvr in nvrs_mapping.items():
    rebuilt_nvr = original_nvr
    while rebuilt_nvr:
        build = ArtifactBuild.query.filter_by(rebuilt_nvr=rebuilt_nvr).first()
        if not build:
            break
        log.info(f"Found former rebuild '{build.rebuilt_nvr}' of '{latest_nvr}'")
        nvr_mappings_to_add[build.original_nvr] = latest_nvr
        rebuilt_nvr = build.original_nvr
```

Reply:
I'm afraid your suggested snippet does not exactly the same thing as the current code.

When I run the test `test_bundle_include_previous_builds`, the code you suggested will return
```python
{
    'original_1': 'some_name-2-12345',
    'original_2': 'some_name_two-2-2'
    'some_name-2-1': 'some_name-2-12345',
}
```
while the current code returns
```python
{
    'original_1': 'some_name-2-12345',
    'original_2': 'some_name_two-2-2',
    'some_name-2-1': 'some_name-2-12345',
    'some_name-2-12344': 'some_name-2-12345',
    'some_name-2-12343': 'some_name-2-12345',
    'some_name-2-12342': 'some_name-2-12345'
}
```
that is, your suggestion does not make the mapping for the previous rebuilds.

Perhaps there is some adjustment that we could do to your suggestion? Because it is much more concise and smaller, I definitely like it better than the current code.

Reply:
I checked the test case, it's missing some db records that's required for the code to look up the original image, for example, we have:
```
'original_1': 'some_name-2-12345'
```
in original nvrs mapping, but there is no db record exists which the `rebuilt_nvr` is `original_1`, we can update the test case to improve the readability and fix the data in db, I added my suggestion to that test case.